recipes-core/disk-encryption/disk-encryption.bb

```
@@ -36,9 +43,14 @@ do_install() {
     install -d ${D}${sysconfdir}/init.d
     install -m 0755 ${WORKDIR}/init ${D}${sysconfdir}/init.d/disk-encryption
 
-    # Create environment file for TARGET_LUN
-    install -d ${D}${sysconfdir}/default
-    echo "TARGET_LUN=${TARGET_LUN}" > ${D}${sysconfdir}/default/disk-encryption
+    # Create environment file for init service
+    install -d ${D}${sysconfdir}/default ${D}${sysconfdir}/disk-encryption
+    echo "TARGET_LUN=${TARGET_LUN}
```

you forgot to write the output to  ${D}${sysconfdir}/default/disk-encryption. Because later it will be sourced from there in the init file, right?
```suggestion
    echo "TARGET_LUN=${TARGET_LUN} > ${D}${sysconfdir}/default/disk-encryption
```

---

It is a multiline `echo`. The redirection to file is on the next line